internal/controller/clusterlogging/clusterlogging_controller.go

```diff
@@ -0,0 +1,163 @@
+	Reader client.Reader
+	Scheme *runtime.Scheme
+}
+
+// +kubebuilder:rbac:groups=logging.openshift.io,resources=clusterloggings,verbs=get;list;create;watch;update;patch;delete
+// +kubebuilder:rbac:groups=logging.openshift.io,resources=clusterloggings/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=logging.openshift.io,resources=clusterloggings/finalizers,verbs=update
+// Reconcile reads that state of the cluster for a ClusterLogging object and makes changes based on the state read
+// and what is in the ClusterLogging.Spec
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileClusterLogging) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.Result, error) {
+	log.V(3).Info("Clusterlogging reconcile request.", "namespace", request.Namespace, "name", request.Name)
+
+	// obs.CLF not found so start migration
+	// Fetch the ClusterLogging instance
+	instance, err := loader.FetchClusterLogging(r.Client, request.NamespacedName.Namespace, request.NamespacedName.Name)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return ctrl.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return ctrl.Result{}, err
+	}
+
+	// Fetch associated logging.ClusterLogForwarder if any
+	clf, err := loader.FetchClusterLogForwarder(r.Client, request.NamespacedName.Namespace, request.NamespacedName.Name)
+	if err != nil && !errors.IsNotFound(err) {
+		return ctrl.Result{}, err
+	}
+
+	// Check if fluentDForward is used & validates output secrets
+	outputSecrets, status, err := clusterlogforwarder.ValidateClusterLogForwarderForConversion(clf, r.Client)
+	if err != nil {
+		clf.Status = *status
+		if result, err := r.updateStatus(instance); err != nil {
+			return result, err
+		}
+		return ctrl.Result{}, err
+	}
+
+	// Convert to observability.ClusterLogForwarder
+	obsClf := api.ConvertLoggingToObservability(r.Client, instance, clf, outputSecrets)
+	// Fix indices for default elasticsearch to be `app-write`, `infra-write`, `audit-write`
+	obsClf.Spec, _ = obsMigrate.MigrateDefaultElasticsearch(obsClf.Spec)
+
+	if err = r.Client.Create(context.TODO(), obsClf); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	// Annotate CL with migrated
+	instance.Annotations[constants.AnnotationCRConverted] = "true"
+	if err = r.Client.Update(context.TODO(), instance); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	// Set and update status
+	instance.Status = loggingv1.ClusterLoggingStatus{}
+	instance.Status.Conditions.SetCondition(conditions.CondReadyWithMessage(loggingv1.ReasonMigrated, "logging.ClusterLogging migrated to observability.ClusterLogForwarder"))
```

Comment:
Maybe make this more specific and correct by saying... ClusterLogForwarder.(logging|observability).openshift.io  since these are the full type of the CRs